### PR TITLE
Allow src dir in run dir

### PR DESCRIPTION
**Pull request by hjoliver**
This is a (very) small change with no associated issue - it just removes a check on source dir location.
 
Currently `cylc install` refuses to install a source directory that is located under `~/cylc-run`, but I don't think there's a good technical reason for that, is there??  We probably just thought it indicated user CLI error or messy housekeeping practices. 

Note there's an additional check that a source dir does not contain `_cylc_install` and `log` dirs etc., which should be sufficient for catching the primary CLI error we're presumably trying to catch (i.e. referring to run directory as a source directory).

Unfortunately the location check prevents a sub-worfklow source dir installed as part of a main workflow, from being used as the `cylc install` source for sub-workflow instances at run time. 

Sub-workflow instances can of course be manually installed by simply copying the source from the main workflow run directory each time, but then we don't get the benefits of `cylc install` and `cylc reinstall`.

With this check disabled, I have everything I need to get (non-nested) sub-workflows working rather nicely, e.g. with the ability to  fix and continue a broken sub-workflow *or* run a new instance of it by re-triggering the main-workflow task - but I'll demo that next week.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Does not need tests
- [x] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

**Comment by MetRonnie** [comment]
> Currently `cylc install` refuses to install a source directory that is located under `~/cylc-run`, but I don't think there's a good technical reason for that, is there?? We probably just thought it indicated user CLI error or messy housekeeping practices.

- #4613
- which addressed #4594

**Review by oliver-sanders — APPROVED** [review]

**Comment by MetRonnie** [comment]
I think a test that you *can* install from a source dir that is inside `~/cylc-run` would be useful

**Comment by hjoliver** [comment]
> I think a test that you _can_ install from a source dir that is inside `~/cylc-run` would be useful.
> 
> And perhaps a test that you can't do that if the source dir was itself `cylc install`ed

:+1:  done

**Comment by hjoliver** [comment]
@MetRonnie - I'll merge this, as the last item on the rc3 milestone. Perhaps you can do a quick post-merge sanity check on the two unit tests added in response to your comment though.

**Review by MetRonnie — COMMENTED** [review]
✔️

File: tests/unit/test_workflow_files.py [hunk 1]
```diff
@@ -1884,12 +1884,19 @@ def test_validate_source_dir(tmp_run_dir: Callable, tmp_src_dir: Callable):
...
         validate_source_dir(src_dir, 'roland')
     assert "exists in source directory" in str(exc_info.value)
-    # Test that src dir is not allowed to be inside ~/cylc-run
+    # Test that src dir is allowed to be inside ~/cylc-run
     src_dir = cylc_run_dir / 'dieter'
     src_dir.mkdir()
+    (src_dir / WorkflowFiles.FLOW_FILE).touch()
+    validate_source_dir(src_dir, 'dieter')
+    # Test that src dir is not allowed to be an installed dir.
+    src_dir = cylc_run_dir / 'ajay'
```
[the remark this answers is not in the document]
Comment (hjoliver) [reply tests/unit/test_workflow_files.py:1893]: Hehe, I hope so :rofl:

**Comment by hjoliver** [comment]
[UPDATE] For the record, this superseded the aborted "support for sub-workflows" PRs #4811 #4821. Allowing source dirs under cylc-run allows proper installation of sub-workflows that belong to a parent workflow; beyond that we're leaving it to external scripts to manage sub-workflow names and run directories.

**Pull request merged**